CONFIGURATION.md

```diff
@@ -103,3 +70,127 @@ agent:
 Note that disabling agent health checks makes is much more difficult for
 Kubernetes to properly assess when the agent is functional (alive) and when the 
 agent is ready to service requests (ready).
+
+### Agent Image Controls
+
+The agent is controlled by a container defintion which pulls the agent's image
+from a container repository. This image contains the spire-agent executable
+which connects to the spire-server through a procedure known as agent-attestation.
+
+The values associated with the agent image include:
+
+| Path                     | Type   | Defaults             |
+| ------------------------ | ------ | -------------------- |
+| agent.image.name         | string | spire/spire-agent    |
+| agent.image.registry     | string | {image.registry}     |
+| agent.image.registryPort | int    | {image.registryPort} |
+| agent.image.tag          | string | {image.tag}          |
+
+> Note: These values default to other values.  If the other values are unset
+> consult thier default values to determine the final value.
+
+When settings are best applied to all images, consider setting them in the the
+[Image Configuration](image_configuration) section.
+
+Here is an example of a customized agent image, where a test agent is being
+used with other non-test components.
+
+```yaml
+agent:
+  image:
+    name: "mycorp/spire-test-agent"
+    registry: "mycorp"
+    registryPort: 8080
+    tag: "latest"
+```
+
+### Agent Key Manager
+
+The agent is configured by default to use a disk KeyManager with a directory
+setting of "/opt/spire/data/agent/".  This setting prevents node reattestation
+for previously attested nodes, as the agent can present previoulsy accquired,
+current SVIDs to attach to the servers.  
+
+This default corresponds to the agent configuration stanza:
+
+```HCL
+    plugins {
+      KeyManager "disk" {
+        plugin_data {
+          directory = "/opt/spire/data/agent/"
+        }
+      }
+    }
+```
+
+These defaults are not for everyone, to change the Agent KeyManager, configure
+the **agent.keyManager.type** setting and apply the settings for the selected
+KeyManager type.
+
+| Path                  | Type   | Default |
+| --------------------- | ------ | ------- |
+| agent.keyManager.type | string | "disk"  |
+
+> Note: **agent.keyManager.type** can only be set to one of "custom", "disk",
+> or "memory".
+
+#### Agent Key Manager - Custom
+
+> Note: These values only effect the configuration when the **agent.keyManager.type**
+> has the value "custom".
+
+| Path                              | Type             | Default            |
+| --------------------------------- | ---------------- | ------------------ |
+| agent.keyManager.custom.checksum  | string           |                    |
+| agent.keyManager.custom.cmd       | string           |                    |
+| agent.keyManager.custom.data      | array of strings | [ ]                |
+| agent.keyManager.custom.name      | string           | "customKeyManager" |
```

Comment:
Should they be required to set the name? Its basically never going to work out of the box.

Reply:
I don't know if the name is really used.  I guess it's an item we will have to do research on.

Reply:
Resovling this one by deferring it into Issue #18